src/test/eo/org/eolang/collections/list-tests.eo

```diff
@@ -601,3 +601,81 @@
           sprintf
             "%d\n"
             x
+
+[] > filteredi-test-1
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 3 1 4 2 5
+        [v i]
+          v.lt 3 > @
+      list
+        * 1 2
+    $.equal-to TRUE
+
+[] > filteredi-test-2
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 7 3 8 4 2 1
+        [v i]
+          v.gt 3 > @
+      list
+        * 7 8 4
+    $.equal-to TRUE
+
+[] > filteredi-test-3
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          *
+        [v i]
+          v.lt 3 > @
+      list
+        *
+    $.equal-to TRUE
+
+[] > filteredi-test-4
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 3 1 4 2 5
+        [v i]
+          i.gt 3 > @
+      list
+        * 5
+    $.equal-to TRUE
+
+[] > filteredi-test-5
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 3 1 4 2 5
+        [v i]
+          v.eq 0 > @
+      list
+        *
+    $.equal-to TRUE
+
+# This code should prints "1 2"
+[] > filteredi-printing-test
+  seq > @
+    filteredi.
+      list
+        * 1 2 3
+      [v i]
+        seq > @
+          if.
+            v.lt 3
+            []
+              stdout > @
+                sprintf
+                  "%d\n"
+                  v
+            FALSE
+    TRUE
```

Comment:
@ivan-egorov42 Can you add a new tests or change the existing one to deal with not only numbers?

Reply:
Yes, of course.